### ISSUE TITLE
Integration is not working at all - IP Allowlist

### ISSUE THREAD

**Issue by n8tr** [atlassian/github-for-jira#763]
# Github for Jira completely broken

- No matter how many times I go through the process, Github for Jira doesn't actually update tickets nor show any connection between Jira and Github in any way at all.
- We do have IP white listing turned on for both JIRA and Github. Do we have to add Github IP addresses to admin.atlassian.com as well?

![image](https://user-images.githubusercontent.com/1009324/134159596-e2ebbd44-7f52-44cc-9538-dea3fedce4b3.png)

### I've gone through this process many times and it still brings me here instead of showing any kind of configuration page:
![image](https://user-images.githubusercontent.com/1009324/134159622-685e91ad-7128-46e8-b6c5-821ad0a8c6fe.png)

### It appears to be set up on Github correctly
![image](https://user-images.githubusercontent.com/1009324/134159935-711f1af3-8e0d-4ee3-9eb6-1cfb296cf8f8.png)

### We do have IP address filtering on for both JIRA and Github, but the JIRA domains do show up on Github
![image](https://user-images.githubusercontent.com/1009324/134160442-3d8c7d0c-f849-4ea4-a4d4-04c86c4b263d.png)

**Comment by rachellerathbone** [comment]
Hey @n8tr. Our team at Atlassian recently migrated the app over from a team a GitHub and we are currently still learning the ins and outs of the app, along with common issues our customers face. One problem that was identified pretty early on was that any customer that allowed any IP whitelisting was unable to use the integration. We have been able to get the app into a state where IP allow listing works from the GH side as long as custom additions are not made to the list. The team is currently trying to determine how we can resolve this issue so it unblocks our users.
 
As for the Jira side - this is not something we have looked into so I can't tell you at this point if this is adding to the problem or not. I'll raise this issue with my team today to see if anyone has any insight into how an IP list in Jira might be impacting the app and will create an issue for follow-up.

**Comment by n8tr** [comment]
Thank you. 
Does that mean you have recently fixed it internally but the updates are not distributed to all users?


FYI: I did try turning off JIRA IP address whitelisting altogether and it did not solve the issue. So that might not be a problem.

**Comment by rachellerathbone** [comment]
No the fix is public but there is still an issue if users add additional IP addresses.

**Comment by n8tr** [comment]
Wouldn't it be impossible to access GitHub if IP whitelisting only has the JIRA domains? In other words, doesn't that basically break all access to GitHub other than the JIRA- 
GitHub app?

**Comment by mboudreau** [comment]
@n8tr Hm, interesting.  What's your jira domain?  We can look through the logs and figure out what's going on.

**Comment by n8tr** [comment]
[here](https://conjoinedtrianglesofsuccess.atlassian.net/) and [email](nate@cmorq.com)

**Comment by mboudreau** [comment]
thanks!  We'll have a look into this and get back to you.

**Comment by n8tr** [comment]
Thank you!

**Comment by n8tr** [comment]
Any luck on this?

**Comment by rachellerathbone** [comment]
@n8tr, @mboudreau currently has a PR open to try and address this. It is currently being reviewed and discussed amongst the team. We'll update again once it has been merged and deployed to prod.

**Comment by bnannier** [comment]
> @n8tr, @mboudreau currently has a PR open to try and address this. It is currently being reviewed and discussed amongst the team. We'll update again once it has been merged and deployed to prod.

It's been open for 15 days, any resolution?

**Comment by mboudreau** [comment]
@n8tr @bnannier Very sorry for the delay.  We've deployed a change to our proxy.  We weren't able to reproduce your issue, so we're not entirely positive if this will fix it, but if you could please try it out now to see if your changes are coming through and please let us know.  Thanks.

**Comment by hrsuzuki** [comment]
This error is also occurring in my environment.

**Comment by n8tr** [comment]
> @n8tr which browser are you using and are you blocking 3rd party cookies? If you are, please enable all cookies.

This is still a problem. And no, I'm not blocking 3rd party cookies. I double-checked the settings in Chrome. It was only blocking 3rd party cookies in incognito mode. I also even tried allowing all cookies in all situations, and that also produced the same exact error "Looks like something went wrong" that I posted above.

**Comment by ysakai4tl4ss** [comment]
@mboudreau I raised a bug ticket regarding this below:
https://jira.atlassian.com/browse/JRACLOUD-77757

The current workaround is to disable GitHub IP Allowlist.

**Comment by n8tr** [comment]
Thank you. I'm not sure that actually qualifies as a workaround, since that just means turn the feature off, which to me just means the feature is 100% broken and there is no workaround.

**Comment by thombergs** [comment]
Hey all. We have identified an issue with GitHub IP allowlists and resolved it yesterday. Could you please let us know if this resolves your issues?

**Comment by n8tr** [comment]
This is ridiculous. I'm the owner of our github organization account and also Jira.

<img width="812" alt="Screen Shot 2021-11-23 at 8 08 21 PM" src="https://user-images.githubusercontent.com/1009324/143153129-287ffc0f-9553-45fc-95ca-a399c6156385.png">

<img width="326" alt="Screen Shot 2021-11-23 at 8 09 31 PM" src="https://user-images.githubusercontent.com/1009324/143153126-0a1c78d3-0378-48bf-bc98-0e1be3a7dfe7.png">
<img width="416" alt="Screen Shot 2021-11-23 at 8 09 27 PM" src="https://user-images.githubusercontent.com/1009324/143153128-f368039d-7b70-416a-afbe-4a9069e7d6ca.png">

**Comment by thombergs** [comment]
Sorry to hear that @n8tr. Could you open up a support ticket with the Atlassian support with a link to this issue? With your customer data, we can look into the logs to see what we can find.

**Comment by ysakai-atlas** [comment]
@hrsuzuki Will you also be able to try connecting again?

**Comment by mboudreau** [comment]
@n8tr Could it be because it's logged into Github with a different account?  We've also deployed a fix for an issue with 3rd party cookies which might have affected you.  If you could try again?

@ysakai-atlas @hrsuzuki Same with you, please try again to see if it's fixing your issue. Cheers.

**Comment by altjx** [comment]
Same exact issue here. Owner of both my Github Repository and Jira/Atlassian account entirely. I don't mean to hijack, but just wanted to provide additional screenshots in case this helps the OP.

As shown in the screenshot, clearly the owner of the Github Organization:

![image](https://user-images.githubusercontent.com/4023423/146646843-a228a3bd-7743-43a8-8d3d-228b986bf681.png)

Further proof of being the owner of all repositroies:

![image](https://user-images.githubusercontent.com/4023423/146646855-5bffa422-1b36-4de1-a3e6-e3c6fed83963.png)

Message from Jira saying I need admin permissions and need to contact myself (the admin):

![image](https://user-images.githubusercontent.com/4023423/146646863-70cdcb27-5a75-4467-babd-df69ddbdf582.png)

**Comment by thombergs** [comment]
Hey all,

we just identified another issue with IP allowlists (also see [this issue](https://github.com/atlassian/github-for-jira/issues/988)). It seems that GitHub's IP allowlists are not evaluated correctly and they block requests in some cases even if the IP addresses are all listed in the allow list.

You can try to add the IP addresses `13.52.5.96` through `13.52.5.111` to your IP allowlist (each IP address individually, not as a CIDR range - it won't work with a CIDR range for some reason). That worked in my tests.

I have opened a support case with GitHub to get a comment from them about why the IP allowlist managed by our app is not working.

Please let me know if that workaround helps in your case. Note that the above IP range is just the range that our servers currently operate from, so it may change in the future, but it would be good to know if this

**Comment by hrsuzuki** [comment]
@thombergs @mboudreau @ysakai-atlas 
Sorry for the late reply.
I tried it again on my own.

The results were the same as the following information described by "n8tr".
https://github.com/atlassian/github-for-jira/issues/763#issuecomment-977353764

The addresses '13.52.5.96' and '13.52.5.111' have been added to the allow list.
Should I open a support ticket again?

**Comment by thombergs** [comment]
@hrsuzuki sorry to hear that. If you want to share your Jira site URL here, I can look into the logs and see what happened. Otherwise, yes, please raise a support case with Atlassian support referencing this issue and request to forward the case to us, and then I'll have a look, too :).

**Comment by n8tr** [comment]
I was able to get the integration to work by disabling the IP allowlist on Github entirely. It did then backfill. I have since re-enabled the IP allowlist, so I'll see if JIRA is still able to connect to Github now that it's back on, or whether it is broken again.

**Comment by thombergs** [comment]
@n8tr we only observed the bug with the allowlist blocking issue with requests that happened during the initial installation of an org, so the workaround you did might just work. I will investigate further and follow up again with GitHub to try to push for a real solution here.

**Comment by n8tr** [comment]
It does appear to be working using the workaround where I turn off the allowlist to get through setup, then turn it back on. I just did a test of a new branch and Jira did recognize the branch and made the association with the ticket. So I think this works!

**Comment by thombergs** [comment]
I just pinged the GitHub support again on the [ticket I raised](https://support.github.com/ticket/personal/0/1429310) concerning the IP allowlist issue. It's been dormant over the holidays, but I'm hoping to hear back from them.

**Comment by hrsuzuki** [comment]
I turned off the "IP allow list" feature during setup and turned it back on after setup and it worked.

Thanks😄 

The procedure is described below.

1. GitHub:  Organization security > IP allow list > 「Enable IP allow list」  And 　「Enable IP allow list configuration for installed GitHub Apps」 turned off.
2. From Jira's application management, use "GitHub for Jira" to collaborate.
3. GitHub:  Organization security > IP allow list > 「Enable IP allow list」  And 　「Enable IP allow list configuration for installed GitHub Apps」 turned on.

**Comment by rachellerathbone** [comment]
🎉 😄 Amazing news @hrsuzuki!

@n8tr please let us know if the above works for you as well.

**Comment by rachellerathbone** [comment]
Appears to be resolved now so closing issue.

**Comment by TomDodson** [comment]
I cannot do the above the solution as I do not subscribe to GitHub Enterprise. IP whitelisting is only included in GitHub Enterprise. Is there a work around for no enterprise users? Thanks!
